File: backend/routes/agentverse.js
import express from 'express';
import { queryAgentverse } from '../lib/agentverse.js';

const router = express.Router();

router.post('/', async (req, res) => {
  const { question, projectContext } = req.body;
  try {
    const response = await queryAgentverse(question, projectContext);
    return res.status(200).json({ result: response });
  } catch (error) {
    return res.status(500).json({ message: error.message || 'Agentverse query failed' });
  }
});

router.all('*', (req, res) => {
  res.status(405).json({ message: 'Method not allowed' });
});

export default router;

File: backend/routes/agentverse.py
from fastapi import APIRouter, HTTPException
from backend.models import AgentverseQueryPayload
from backend.lib.agentverse import query_agentverse

router = APIRouter()

@router.post('/')
def ask_agentverse(payload: AgentverseQueryPayload):
    try:
        response = query_agentverse(payload.question, payload.projectContext)
        return {'result': response}
    except Exception as exc:
        raise HTTPException(status_code=500, detail=str(exc))
